frontend

ulist2020 opened the pull request.

ulist2020 committed:
fix: Установка CRA

ulist2020 committed:
fix: Компоненты и файловая структура

ulist2020 committed:
fix: подготовить роуты

ulist2020 committed:
fix: сверстать хедер

ulist2020 committed:
fix: сверстать хедер + навигация

ulist2020 committed:
fix: сделать адаптив хедер + навигация

ulist2020 committed:
fix: сверстать промо

ulist2020 committed:
fix: сверстать блок о проекте

ulist2020 committed:
fix: сверстать блок технологии

ulist2020 committed:
fix: сверстать блок обо мне

ulist2020 committed:
fix: сверстать блок портфолио

ulist2020 committed:
fix: сверстать футер

ulist2020 committed:
fix: сверстать блок найти фильм + чекбокс

ulist2020 committed:
fix: сверстать страницу со списком фильмов

ulist2020 committed:
fix: сверстать блок сохраненные фильмы

ulist2020 committed:
fix: сверстать регистрацию

ulist2020 committed:
fix: сверстать регистрацию и логин

ulist2020 committed:
fix: исправить хедер

ulist2020 committed:
fix: исправить блоки регистрации и логин

ulist2020 committed:
fix: сверстать страницу профайл

ulist2020 committed:
fix: сверстать страницу 404

ulist2020 committed:
fix: сверстать попап

ulist2020 committed:
fix: сделать логотип ссылкой + скрол

ulist2020 committed:
Update README.md

ulist2020 committed:
Update README.md

ulist2020 committed:
создать папку билд

coolswood commented:
# Резюме по работе:
### Критерии, влияющие на работоспособность
- [x] <!--10.71--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [ ] <!--10.71--> Проект соответствует сгенерированному под него макету.
`В бургерном меню отсутствует кнопка "аккаунт". Поля для заполнения на странице регистрации слишком длинные.`
<img width="639" alt="Снимок экрана 2021-08-10 в 00 48 47" src="https://user-images.githubusercontent.com/25247802/128779144-9678c222-c3c9-4429-a8b2-ed3a228fd91f.png">
<img width="690" alt="Снимок экрана 2021-08-10 в 00 47 25" src="https://user-images.githubusercontent.com/25247802/128779222-a64984ba-b0d4-4f75-ae36-e2518d4dd02c.png">

- [ ] <!--10.71--> Все блоки из макета свёрстаны. Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
`Нет двух состояний кнопки в карточке`
<img width="253" alt="Снимок экрана 2021-08-10 в 00 49 38" src="https://user-images.githubusercontent.com/25247802/128779092-a0bd96fd-d44f-40a0-a5df-a61eb4ebbc9e.png">
<img width="252" alt="Снимок экрана 2021-08-10 в 00 49 46" src="https://user-images.githubusercontent.com/25247802/128779095-fa5b4fa5-86b2-4933-b44a-a62005a6dda5.png">

- [x] <!--10.71--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
- [x] <!--2.86--> Отсутствуют ошибки валидации при сборке приложения или в валидаторе.
- [x] <!--2.86--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
- [x] <!--2.86--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--2.86--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--2.86--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--2.86--> Работа модальных окон настроена.
- [x] <!--2.86--> Разметка портирована в JSX:x
- [x] <!--2.86--> разметка заключена в ( ), лишние пустые <div> для этого не используются;
- [x] <!--2.86--> вынесена в соответствующие ей компоненты.
- [x] <!--2.86--> В проекте есть:
- [x] <!--2.86--> директория images/ с картинками,
- [x] <!--2.86--> директория components/ с JS и CSS-файлами компонентов,
- [x] <!--2.86--> директория fonts/ со шрифтами.

### Хорошие практики
- [x] <!--2.14--> Именование классов сделано по БЭМ.
- [x] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> reset.css запрещен;
- [x] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
- [x] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.

### Рекомендации
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;

## Количество баллов: 79

ulist2020 commented:
Исправлены замечания после первой проверки.

ulist2020 committed:
Поправить вёрстку.

Добавить активный лайк, добавить кнопку удаления сохраненного фильма, исправить размер инпутов.

coolswood commented:
# Резюме по работе:
### Критерии, влияющие на работоспособность
- [ ] <!--10.71--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
`После устранения ошибки длинных полей страницы регистрации появился горизонтальный скролл на расширении от 1200рх до 900рх`
- [x] <!--10.71--> Проект соответствует сгенерированному под него макету.


- [x] <!--10.71--> Все блоки из макета свёрстаны. Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
`Нет двух состояний кнопки в карточке`
<img width="253" alt="Снимок экрана 2021-08-10 в 00 49 38" src="https://user-images.githubusercontent.com/25247802/128779092-a0bd96fd-d44f-40a0-a5df-a61eb4ebbc9e.png">
<img width="252" alt="Снимок экрана 2021-08-10 в 00 49 46" src="https://user-images.githubusercontent.com/25247802/128779095-fa5b4fa5-86b2-4933-b44a-a62005a6dda5.png">

- [ ] <!--10.71--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
`Кнопка аккаунт съезжает за всплывающее меню`
<img width="276" alt="Снимок экрана 2021-08-11 в 11 29 49" src="https://user-images.githubusercontent.com/25247802/128997090-5a5233e9-40d0-47a7-bcd8-615808899a65.png">

- [x] <!--2.86--> Отсутствуют ошибки валидации при сборке приложения или в валидаторе.
- [x] <!--2.86--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
- [x] <!--2.86--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--2.86--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--2.86--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--2.86--> Работа модальных окон настроена.
- [x] <!--2.86--> Разметка портирована в JSX:x
- [x] <!--2.86--> разметка заключена в ( ), лишние пустые <div> для этого не используются;
- [x] <!--2.86--> вынесена в соответствующие ей компоненты.
- [x] <!--2.86--> В проекте есть:
- [x] <!--2.86--> директория images/ с картинками,
- [x] <!--2.86--> директория components/ с JS и CSS-файлами компонентов,
- [x] <!--2.86--> директория fonts/ со шрифтами.

### Хорошие практики
- [x] <!--2.14--> Именование классов сделано по БЭМ.
- [x] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> reset.css запрещен;
- [x] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
- [x] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.

### Рекомендации
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;

## Количество баллов: 79

ulist2020 committed:
исправить маржены

ulist2020 commented:
Исправлены замечания после второй проверки⬆

coolswood commented:
# Резюме по работе:
### Критерии, влияющие на работоспособность
- [x] <!--10.71--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
`После устранения ошибки длинных полей страницы регистрации появился горизонтальный скролл на расширении от 1200рх до 900рх`
- [x] <!--10.71--> Проект соответствует сгенерированному под него макету.


- [x] <!--10.71--> Все блоки из макета свёрстаны. Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
`Нет двух состояний кнопки в карточке`
<img width="253" alt="Снимок экрана 2021-08-10 в 00 49 38" src="https://user-images.githubusercontent.com/25247802/128779092-a0bd96fd-d44f-40a0-a5df-a61eb4ebbc9e.png">
<img width="252" alt="Снимок экрана 2021-08-10 в 00 49 46" src="https://user-images.githubusercontent.com/25247802/128779095-fa5b4fa5-86b2-4933-b44a-a62005a6dda5.png">

- [x] <!--10.71--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
`Кнопка аккаунт съезжает за всплывающее меню`
<img width="276" alt="Снимок экрана 2021-08-11 в 11 29 49" src="https://user-images.githubusercontent.com/25247802/128997090-5a5233e9-40d0-47a7-bcd8-615808899a65.png">

- [x] <!--2.86--> Отсутствуют ошибки валидации при сборке приложения или в валидаторе.
- [x] <!--2.86--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
- [x] <!--2.86--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--2.86--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--2.86--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--2.86--> Работа модальных окон настроена.
- [x] <!--2.86--> Разметка портирована в JSX:x
- [x] <!--2.86--> разметка заключена в ( ), лишние пустые <div> для этого не используются;
- [x] <!--2.86--> вынесена в соответствующие ей компоненты.
- [x] <!--2.86--> В проекте есть:
- [x] <!--2.86--> директория images/ с картинками,
- [x] <!--2.86--> директория components/ с JS и CSS-файлами компонентов,
- [x] <!--2.86--> директория fonts/ со шрифтами.

### Хорошие практики
- [x] <!--2.14--> Именование классов сделано по БЭМ.
- [x] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> reset.css запрещен;
- [x] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
- [x] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.

### Рекомендации
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;

## Количество баллов: 100
